### DISCUSSION TITLE
Any support for JSON mode coming soon?

### DISCUSSION THREAD

**Discussion by hasani114** [open-webui/open-webui#85] [Ideas]
Ollama just introduced a json mode on their end, do you have any plans on adding a property on the webui that will allow for this to be toggled?

**Comment by tjbck** [comment]
I'm currently ideating ways I could incorporate it, most likely It'll be added to one of the settings menu. I'll keep you updated!

**Comment by tjbck** [comment]
Implemented with #86, you can now toggle the request mode from default to JSON (and vice versa) from settings > advanced. Thanks.

**Comment by hasani114** [comment]
Thanks!

**Comment by kinthaiofficial** [comment]
JSON mode matters a lot once you move from "chatbot" to "agent-enabled UI" — structured outputs are what let the UI reliably parse model responses and take actions (render a chart, populate a form, trigger an API call).

A few things worth getting right from the start:

**JSON mode vs schema-constrained output** — JSON mode (requesting valid JSON without a schema) is useful for simple cases but breaks down when the UI needs a specific structure. Schema-constrained output (via tools/functions or guided decoding) is what you actually want for agentic UIs. The schema is the contract between the model and the UI renderer.

**Streaming JSON** — if responses are long, you want to stream the JSON and render incrementally (show the first fields while the rest generate). Proper streaming JSON support means parsing partial JSON gracefully and updating the UI as more fields arrive.

**Fallback for non-compliant responses** — not all models (especially local ones) produce valid JSON reliably. The UI needs a graceful fallback: show the raw text if JSON parsing fails, rather than crashing or showing nothing.

**Tool calling as the better abstraction** — for agent-capable UIs, tool calls (OpenAI-style or A2A-style) are more robust than JSON mode. The model explicitly signals "I want to take this action" rather than "here's some JSON hope it parses."

We've been building agent-facing UIs in KinthAI where the UI is itself an agent — it can spawn sub-agents, display their progress, and handle budget/cost display. Write-up on the design: https://blog.kinthai.ai/221-agents-multi-agent-coordination-lessons

Is the JSON mode request primarily for displaying structured data, or for triggering actions based on model output?
